libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/DataStore.java

```diff
@@ -154,14 +157,49 @@ public T delete(String partitionValue, String sortValue) {
         return table.deleteItem(key);
     }
 
-    public List<T> delete(List<T> items) {
-        return items.stream().map(table::deleteItem).toList();
-    }
-
+    @ExcludeFromGeneratedCoverageReport
     public void deleteAllByPartition(String partitionValue) {
         delete(getItems(partitionValue));
     }
 
+    @ExcludeFromGeneratedCoverageReport
+    public int delete(List<T> items) {
+        if (!items.isEmpty()) {
+            BatchWriteResult batchWriteResult =
+                    processBatchWrite(createWriteBatchForDeleteItems(items));
+            // 'unprocessedDeleteItemsForTable()' returns keys for delete requests that did not
+            // process.
+            List<Key> unprocessedItems =
+                    batchWriteResult.unprocessedDeleteItemsForTable(this.table);
+            if (!unprocessedItems.isEmpty()) {
+                unprocessedItems.forEach(
+                        key ->
+                                LOGGER.warn(
+                                        "Delete fail for item key: %s".formatted(key.toString())));
```

Comment:
Do you think this is a warning or an error? In the 'end of journey' case, a warning is fine, but I think for the in-session reset case, this is an error - attempting to continue might cause unexpected results.

I'd probably just throw an exception?

Reply:
If we throw an exception (as needed by in-session reset cse)) here then I suppose this delete() method don't need to return int.